game-core/src/main/java/games/strategy/engine/lobby/client/login/CreateAccountPanel.java

```diff
@@ -195,19 +198,75 @@
             0,
             0));
 
+    final JLabel validationLabel = new JLabel(" ");
+    validationLabel.setForeground(Color.RED);
+
+    main.add(
+        validationLabel,
+        new GridBagConstraints(
+            0,
+            5,
+            2,
+            1,
+            0.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(5, 5, 0, 0),
+            0,
+            0));
+
     final JPanel buttons = new JPanel();
     add(buttons, BorderLayout.SOUTH);
     buttons.setBorder(BorderFactory.createEmptyBorder(10, 5, 10, 5));
     buttons.setLayout(new FlowLayout(FlowLayout.RIGHT, 5, 0));
     final JButton okButton = new JButton("OK");
+    okButton.setEnabled(false);
     buttons.add(okButton);
     final JButton cancelButton = new JButton("Cancel");
     buttons.add(cancelButton);
 
     cancelButton.addActionListener(e -> close());
-    okButton.addActionListener(e -> okPressed());
+    okButton.addActionListener(
+        e -> {
+          if (okButton.isEnabled()) {
+            okPressed();
+          }
+        });
+
+    SwingComponents.addEnterKeyListener(
+        this,
+        () -> {
+          if (okButton.isEnabled()) {
+            okPressed();
```

Comment:
Would it make sense to move this if check inside the `okPressed` method?
I know that you don't like methods silently being no-ops if some certain condition is true, but I think if the method is properly renamed this would be worth it to reduce code-duplication?

Reply:
Fixed in: https://github.com/triplea-game/triplea/pull/5368/commits/956d225394a68cc1ddd944422b3a4ce1213e275a

Reply:
Definitely an improvement, even though I still think it's weird checking if the button is enabled before "triggering a button pressed event".
I really feel like there should be a built-in way to link certain textfields to a button so this check would be performed automatically

Reply:
Maybe the original problem is that the 'okButtonPressed' method has actually nothing to do with the button. Arguably it's badly named as its named after the implementation that triggers the action and not after the action itself. Hence, the 'enter' key press is mapped to the same action, bypasses the button altogether. The button enabled is our state tracking if the fields are valid or not...

> I really feel like there should be a built-in way to link certain textfields to a button so this check would be performed automatically

Not that I'm aware of. KeyTypedListener is about as close as it gets. In the swing rewrite of the launch screen there were a few components like this that were created FWIW..